proto/p2p/p2p.proto

```diff
@@ -385,7 +386,8 @@ message Chits {
 
 // AppRequest is a VM-defined request.
 //
-// Remote peers must respond to AppRequest with corresponding AppResponse
+// Remote peers must respond to AppRequest with a corresponding AppResponse or
+// AppRequestFailed
```

Comment:
From the perspective of the networking spec:

- Requester: if the requester sees a timeout from the remote peer it must queue up an AppRequestFailed message from the responder peer to time out the request
- Responder: If the responder detects that it has timed out, the behavior here does not matter because any response will be dropped by the requester. 

So I guess both are technically valid, but I think (2) of just moving on would be best since clients should have the final responsibility for how their pending requests are maintained.